### ISSUE TITLE
run --entrypoint not working as expected

### ISSUE THREAD

**Issue by jareware** [docker/compose#1618]
Given this `docker-compose.yml`:
```
test:
  image: node:0.10.35
```

The following works as expected:
```
$ docker run --entrypoint bash node:0.10.35 -c 'echo HELLO'
HELLO
```

But the following doesn't:
```
$ docker-compose run --entrypoint bash test -c 'echo HELLO'
Cannot start container d160893c56b9f3c0f60e7ace1f0a004095eecb5b40a0fee14a287e265939f138: [8] System error: exec: "\"bash\"": executable file not found in $PATH
```

My versions are:
```
$ docker -v
Docker version 1.7.0, build 0baf609
$ docker-compose -v
docker-compose version: 1.3.1
CPython version: 2.7.9
OpenSSL version: OpenSSL 1.0.1e 11 Feb 2013
```

**Comment by volcomism** [comment]
Likely relates to: https://github.com/docker/docker/issues/14170

**Comment by jareware** [comment]
Looks that way! In the meantime, worked around this simply by downgrading to:
```
Docker version 1.5.0, build a8a31ef
docker-compose 1.1.0
```

**Comment by TamerShlash** [comment]
@jareware how did you downgrade to docker-compose 1.1.0?

**Comment by aanand** [comment]
You don't have to downgrade Compose, just Docker - Compose 1.3.0 will work just fine with Docker 1.6.2.

**Comment by aanand** [comment]
Docker 1.7.1 RC1 is out:

https://test.docker.com/builds/Linux/x86_64/docker-1.7.1-rc1
https://test.docker.com/builds/Darwin/x86_64/docker-1.7.1-rc1

**Comment by dnephin** [comment]
This was fixed in docker 1.7.1
